### DISCUSSION TITLE
Instance members of Source doesn't have a type definition.

### DISCUSSION THREAD

**Discussion by adellamaggiora** [maplibre/maplibre-gl-js#1855] [Q&A]
[https://maplibre.org/maplibre-gl-js-docs/api/sources/#geojsonsource-instance-members](url)

Property 'getClusterExpansionZoom' does not exist on type 'Source'.
Property 'getClusterChildren' does not exist on type 'Source'.
Property 'getClusterLeaves' does not exist on type 'Source'.
Property 'setData' does not exist on type 'Source'.

and this causes a TypeScript compiler error.

**Comment by HarelM** [comment]
Please cast source to geojsonsource.

**Reply by adellamaggiora** [reply]
Ok, I could cast to any too xD.
Are you planning to fix this signature? thx

**Reply by HarelM** [reply]
The source that is returned can be any source , geojson, vector, image etc.
You need to cast it to the relevant type.
Casting to any won't allow you auto complete...

**Comment by mradamcox** [comment]
I'm finding this same problem and don't understand how to cast the returned Source as GeoJSONSource. 

```
const features = map.queryRenderedFeatures(e.point, {
      layers: [lyrDef.spec.id]
});
const clusterId = features[0].properties.cluster_id;
const zoom = await map.getSource("sourceid").getClusterExpansionZoom(clusterId);
map.easeTo({
      center: features[0].geometry.coordinates,
      zoom
});
```

As far as I can tell this mimics the official docs here: https://docs.mapbox.com/mapbox-gl-js/api/sources/#geojsonsource#getclusterexpansionzoom, however, I'm getting a type error on `.getClusterExpansionZoom()` saying 

```
Property 'getClusterExpansionZoom' does not exist on type 'Source'.ts(2339)
```

What is the proper way to do this?

For what it's worth, I also get a type error on `.coordinates` saying it doesn't exist on type `Geometry` or `GeometryCollection`, but I don't know if that's related or something else.

**Reply by mirzoibodullaev** [reply]
`
const coordinates = (features[0].geometry as GeoJSON.Point)
            .coordinates as [number, number];
mapInstance.easeTo({
            center: coordinates,
            zoom,
});
`
